fdbserver/TLogServer.actor.cpp

```diff
@@ -1531,7 +1531,7 @@ std::deque<std::pair<Version, LengthPrefixedStringRef>>& getVersionMessages(Refe
 ACTOR Future<Void> waitForMessagesForTag(Reference<LogData> self, TLogPeekRequest* req, double timeout) {
 	self->blockingPeeks += 1;
 	auto tagData = self->getTagData(req->tag);
-	if (tagData.isValid()) {
+	if (tagData.isValid() && !tagData->versionMessages.empty() && tagData->versionMessages.back().first > req->begin) {
```

Comment:
Should it be `tagData->versionMessages.back().first >= req->begin`?

Reply:
Yes, I think you're right thanks! I will fix this